xmppserver/src/main/java/org/jivesoftware/openfire/net/RespondingServerStanzaHandler.java

```diff
@@ -84,6 +90,12 @@ private static boolean isDialbackOffered(Element doc) {
         return doc.element("dialback") != null;
     }
 
+    /**
```

Comment:
Is there another solution to implement this functionality, other than overriding this method?

The original method definition was very simple: check if a provided value defines the start of a stream. With the changes in this override, there is a _lot_ of additional, almost implicit functionality being added.

This typically leads to weirdly interdependant code, that's hard to build further upon and hard to maintain. Can we somehow move the functionality that's now added _in_ this method, to something that's invoked based on the outcome of the method invocation?

Reply:
Agree - I've moved this code to the initiate stream method.